File: training101919/UVa264.cpp
#include <bits/stdc++.h>

using namespace std;

string f( int n ){
  int l = (-1 + sqrt( 1 + 8 * n )) / 2; // get max val for the sequence
  int steps = n - l * l;
  int b = (l + 1) * (l + 1) - steps, a = 1 + steps;
  
}

int main(){
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  int n {};
  while( cin >> n ){
    cout << "TERM " << n << " IS " << f( n ) << '\n';
  }
  return 0;
}

File: training101919/UVa264.py
import sys, fractions, math


line = sys.stdin.readline()
while line:
    n = int(line)
    l = int((-1 + math.sqrt( 1 + 8 * n ) ) // 2)
    r = l + 1
    steps = n - l
    b, a = max(l - steps, 1), 1 + steps
    print('n', n, 'l', l, 'r', r, 'steps', steps, 'b', b, 'a', a)
    ans = fractions.Fraction(a, b)
    print('THE', n, 'TERM IS', str(ans))
    line = sys.stdin.readline()
